edi_oca/models/edi_exchange_record.py

```diff
@@ -244,6 +260,18 @@ def action_exchange_process(self):
         self.ensure_one()
         return self.backend_id.exchange_process(self)
 
+    def action_retry(self):
+        if not self.retryable:
+            return False
+        new_state = self._rollback_state_mapping[self.edi_exchange_state]
+        fname = "edi_exchange_state"
+        self[fname] = new_state
+        display_state = self._fields[fname].convert_to_export(self[fname], self)
+        self.message_post(
+            body=_("Action retry: state moved back to '%s'") % display_state
+        )
+        return True
```

Comment:
> With this new action, I think we should not process failed files by default here: https://github.com/OCA/edi/blob/14.0/edi_oca/models/edi_backend.py#L543.

I don't get this, can you elaborate? From my POV if the file failed either the system or the user can try it again. What is probably desiderable is that that system stops after N attempts to try retry them?

> Also, it would be good if it accepts recordsets, to be potentially used in batch with a server actio

Sure, I'll change it.